test/subtract_avg_cfl_test.cc

```diff
@@ -0,0 +1,142 @@
+/*
+ * Copyright(c) 2019 Netflix, Inc.
+ * SPDX - License - Identifier: BSD - 2 - Clause - Patent
+ */
+
+/*
+ * Copyright (c) 2016, Alliance for Open Media. All rights reserved
+ *
+ * This source code is subject to the terms of the BSD 2 Clause License and
+ * the Alliance for Open Media Patent License 1.0. If the BSD 2 Clause License
+ * was not distributed with this source code in the LICENSE file, you can
+ * obtain it at www.aomedia.org/license/software. If the Alliance for Open
+ * Media Patent License 1.0 was not distributed with this source code in the
+ * PATENTS file, you can obtain it at www.aomedia.org/license/patent.
+ */
+
+/******************************************************************************
+ * @file subtract_avg_cfl_test.cc
+ *
+ * @brief Unit test for cfl subtract average function:
+ * - subtract_average_avx2
+ *
+ * @author Cidana-Edmond
+ *
+ ******************************************************************************/
+
+#include "gtest/gtest.h"
+#include "EbIntraPrediction.h"
+#include "aom_dsp_rtcd.h"
+#include "random.h"
+#include "util.h"
+
+/**
+ * @brief Unit test for cfl subtract average function:
+ * - subtract_average_avx2
+ *
+ * Test strategy:
+ * Verify this assembly code by comparing with reference c implementation.
+ * Feed the same data and check test output and reference output.
+ *
+ * Expected result:
+ * Output from assemble functions should be the same with output from c.
+ *
+ * Test coverage:
+ * Test cases:
+ * data buffer: Fill with random values
+ * TxSize: all the TxSize.
+ */
+
+#define MAKE_TEST_PARAM(function)                                            \
+    {                                                                        \
+        make_tuple(TX_4X4, &function), make_tuple(TX_4X8, &function),        \
+            make_tuple(TX_4X16, &function), make_tuple(TX_8X4, &function),   \
+            make_tuple(TX_8X8, &function), make_tuple(TX_8X16, &function),   \
+            make_tuple(TX_8X32, &function), make_tuple(TX_16X4, &function),  \
+            make_tuple(TX_16X8, &function), make_tuple(TX_16X16, &function), \
+            make_tuple(TX_16X32, &function), make_tuple(TX_32X8, &function), \
+            make_tuple(TX_32X16, &function), make_tuple(TX_32X32, &function) \
+    }
+
+namespace {
+
+using std::make_tuple;
+using svt_av1_test_tool::SVTRandom;
+
+/** Defines of test parameters */
+typedef cfl_subtract_average_fn (*get_sub_avg_fn)(TxSize tx_size);
+typedef std::tuple<TxSize, get_sub_avg_fn> sub_avg_param;
+
+class CflSubAvgTest : public ::testing::TestWithParam<sub_avg_param> {
+  public:
+    CflSubAvgTest() : tx_size_(TEST_GET_PARAM(0)) {
+        width_ = tx_size_wide[tx_size_];
+        height_ = tx_size_high[tx_size_];
+        memset(data_, 0, sizeof(data_));
+        memset(data_ref_, 0, sizeof(data_ref_));
+
+        /** get test and reference function */
+        sub_avg = TEST_GET_PARAM(1)(tx_size_);
```

Comment:
If get_subtract_average_fn_avx2() is the only test target, we can assign function ptr by get_subtract_average_fn_avx2 directly. And remove function out of MAKE_TEST_PARAM macro.

Reply:
it's done